project-koku/koku
# [COST-1142] Exact filter matching

## Jira Ticket

[COST-1142](https://issues.redhat.com/browse/COST-1142)

## Description

This change will add the filter option to exact match on data types

## Testing

1. Checkout Branch
2. Restart Koku
3. make create/load test customer data
4. Go to any API endpoint and try adding filter[exact:something]=EXACT_NAME
5. The returned data should only include exact matches 

## Notes
Project match example: http://localhost:8000/api/cost-management/v1/reports/openshift/costs/?group_by[project]=*&filter[exact:project]=openshift-ocp_prod

Side point, this DOES NOT add support for Exact tag matching. Thats a bigger feat and we should create a new issue specifically if we want to try and tackle that.

This also misses out project support on the OCP on ALL endpoint. For some reason that endpoint does not default to the `OCPALLCostLineItemProjectDailySummaryP` table, this means the namespace filtering fails due to the table having an array. We _could_ switch to this table instead, BUT there is a bunch of fields missing that will likely break other things. We can add support later if we want but again I suggest creating a follow up issue since its going to be a large piece of work. Plus the UI today doesn't even make use of these anyway.
...
